src/search-modal/SearchResult.jsx

```diff
@@ -1,37 +1,151 @@
+import { getConfig, getPath } from '@edx/frontend-platform';
+import {
+  Button,
+  Icon,
+  IconButton,
+  Stack,
+} from '@openedx/paragon';
+import {
+  Article,
+  Folder,
+  OpenInNew,
+  Question,
+  TextFields,
+  Videocam,
+} from '@openedx/paragon/icons';
+import {
+  Highlight,
+  Snippet,
+} from 'react-instantsearch';
+import { useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+
+import { getStudioHomeData } from '../studio-home/data/selectors';
 
 /**
- * A single search result (row), usually represents an XBlock/Component
- * @type {React.FC<{hit: import('instantsearch.js').Hit<{
- *   id: string,
- *   display_name: string,
- *   block_type: string,
- *   'content.html_content'?: string,
- *   'content.capa_content'?: string,
- *   breadcrumbs: {display_name: string}[]}>,
+  * @typedef {import('instantsearch.js').Hit<{
+  *  id: string,
+  *  usage_key: string,
+  *  context_key: string,
+  *  display_name: string,
+  *  block_type: string,
+  *  'content.html_content'?: string,
+  *  'content.capa_content'?: string,
+  *  breadcrumbs: {display_name: string}[]
+  *  breadcrumbsNames: string[],
+  *  }>} CustomHit
+  */
+
+/**
+ * Custom Highlight component that uses the <b> tag for highlighting
+ * @type {React.FC<{
+ *   attribute: keyof CustomHit | string[],
+ *   hit: CustomHit,
+ *   separator?: string,
  * }>}
  */
-const SearchResult = ({ hit }) => (
-  <div className="my-2 pb-2 border-bottom">
-    <div className="hit-name small">
-      <strong><Highlight attribute="display_name" hit={hit} /></strong>{' '}
-      (<BlockTypeLabel type={hit.block_type} />)
-    </div>
-    <div className="hit-description x-small text-truncate">
-      <Highlight attribute="content.html_content" hit={hit} />
-      <Highlight attribute="content.capa_content" hit={hit} />
-    </div>
-    <div className="text-muted x-small">
-      {hit.breadcrumbs.map((bc, i) => (
-        // eslint-disable-next-line react/no-array-index-key
-        <span key={i}>{bc.display_name} {i !== hit.breadcrumbs.length - 1 ? '/' : ''} </span>
-      ))}
-    </div>
-  </div>
+const CustomHighlight = ({ attribute, hit, separator }) => (
+  <Highlight
+    attribute={attribute}
+    hit={hit}
+    separator={separator}
+    highlightedTagName="b"
+  />
 );
 
+const ItemIcon = {
+  vertical: Folder,
+  sequential: Folder,
+  chapter: Folder,
+  problem: Question,
+  video: Videocam,
+  html: TextFields,
+};
+
+/**
+ * Returns the URL for the context of the hit
+ * @param {CustomHit} hit
+ * @param {boolean} newWindow
+ * @param {string} libraryAuthoringMfeUrl
+ * @returns {string}
+ */
+const getContextUrl = (hit, newWindow, libraryAuthoringMfeUrl) => {
+  const { context_key: contextKey, usage_key: usageKey } = hit;
+  if (contextKey.startsWith('course-v1:')) {
+    if (newWindow) {
+      return `${getPath(getConfig().PUBLIC_PATH)}/course/${contextKey}?show=${encodeURIComponent(usageKey)}`;
```

Comment:
And finally: [Figma says](https://www.figma.com/file/DmUC2e7BAoQsGDq4KgApPa/Content-Tagging-MVP?type=design&node-id=14449-36944&mode=design&t=VeD5TOqFeNIOaN15-4) that when we click on a component (text, video, etc), we should be directed to the Unit edit page with the component highlighted. instead of just trying to scroll down to the Unit in the outline.

(I also think searching for a Unit should take us to the Unit edit page too, but that's up to UI/UX, see above :)

Reply:
> Also, since these returned URLs share a common path string, can that be reflected in the code too?

Sure! Done here: b71b0b5dbe11d1beb6bb71d6816bb00ecbd8b271

Reply:
> And finally: [Figma says](https://www.figma.com/file/DmUC2e7BAoQsGDq4KgApPa/Content-Tagging-MVP?type=design&node-id=14449-36944&mode=design&t=VeD5TOqFeNIOaN15-4) that when we click on a component (text, video, etc), we should be directed to the Unit edit page with the component highlighted. instead of just trying to scroll down to the Unit in the outline.

Ack! We will need a follow-up task to that because right now, we don't have the unit info in the component result.